lib/storage/data/external/AwsClient.js

```diff
@@ -427,21 +446,20 @@ class AwsClient {
             },
         };
         const completeObjData = { key: awsKey };
-        return this._client.completeMultipartUpload(mpuParams,
-            (err, completeMpuRes) => {
-                if (err) {
-                    if (mpuError[err.code]) {
+        return this._client.send(new CompleteMultipartUploadCommand(mpuParams)).catch(err => {
+
+                    if (mpuError[err.name]) {
                         logHelper(log, 'trace', 'err from data backend on ' +
                     'completeMPU', err, this._dataStoreName, this.clientType);
-                        return callback(errors[err.code]);
+                        return callback(errors[err.name]);
                     }
                     logHelper(log, 'error', 'err from data backend on ' +
                 'completeMPU', err, this._dataStoreName, this.clientType);
                     return callback(errorInstances.ServiceUnavailable
                         .customizeDescription('Error returned from ' +
                   `${this.type}: ${err.message}`),
                     );
-                }
+                }).then(completeMpuRes => {
```

Comment:
Better to keep the .then before the .catch here I think, same reason as above comment